runtime/src/lib.rs

```diff
@@ -646,7 +650,11 @@ impl PrivilegeCmp<OriginCaller> for OriginPrivilegeCmp {
                     r_yes_votes,
                     r_count,
                 )), // Equivalent to (l_yes_votes / l_count).cmp(&(r_yes_votes / r_count))
-            ) => Some((l_yes_votes * r_count).cmp(&(r_yes_votes * l_count))),
+            ) => Some(
+                l_yes_votes
+                    .saturating_mul(*r_count)
+                    .cmp(&r_yes_votes.saturating_mul(*l_count)),
```

Comment:
```suggestion
                    .cmp(&r_yes_votes.saturating_mul(l_count)),
```

Should not be a deference, I believe

Reply:
That's incorrect, both `left` and `right` are references, and so when they're matched against, all their internal fields are also references. If I don't dereference, I'd be multiplying a reference here, not the actual number.

All saturating math functions aren't like the regular math operators where there's an auto-deref of the operands other than the operand on the callee, so this deref is necessary.

More importantly, this is the kind of thing that the Rust compiler catches, so if your belief is true, then I would expect the compiler to throw errors as it is now.

Reply:
yeah CI would definitely catch this if it was incorrect